lib/ansible/modules/network/nxos/nxos_interface.py

```diff
@@ -56,6 +56,27 @@
         This option is supported for ethernet and portchannel interface.
         Applicable for ethernet and portchannel interface only.
     choices: ['layer2','layer3']
+  bfd:
+    description:
+      - Enable/Disable Bidirectional Forwarding Detection (BFD) on the interface.
+      - "Dependency: feature bfd"
+    version_added: 2.9
+    type: str
+    choices: ['enable', 'disable']
+  bfd_echo:
+    description:
+      - Enable/Disable BFD Echo functionality on the interface.
+      - "Dependency: feature bfd"
+    version_added: 2.9
+    type: str
+    choices: ['enable', 'disable']
+  hsrp_bfd:
+    description:
+      - Enable/Disable HSRP BFD functionality on the interface.
+      - "Dependency: feature bfd, feature hsrp"
+    version_added: 2.9
+    type: str
+    choices: ['enable', 'disable']
```

Comment:
Instead of having three different parameters, can we add these parameteres as options for bfd? 
bfg: {state: '', echo: '', hsrp: ''}

Reply:
I like your idea but it may not be a good fit in this case. My concern is that there are other hsrp and bfd options to consider.

With hsrp interface options, the bfd state is just one of several possible hsrp options and it seems like it would be better to associate the `hsrp bfd` state with `hsrp` options:
```
n9k-109(config-if)# hsrp ?
  <0-4095>     Group number (0-255 for HSRPv1)
  bfd          BFD protocol
  delay        HSRP initialisation delay
  mac-refresh  Interface mac-refresh time
  use-bia      HSRP uses interface's burned in address
  version      HSRP version
```

For `bfd` interface options, several of them mimic the bfd global settings (to allow per-interface tweaking) so I would expect them to use the same convention as in `bfd_global`. A case can be made for `bfd_options: {bfd: '', echo: ''}` but those two options would be exceptions to the standalone option format used with the other bfd options. (*which may be fine, I just wanted you to be aware before we change anything*)

global:
```
n9k-109(config)# bfd ?
  echo-interface    Configure interface used for bfd echo frames
  echo-rx-interval  Configure BFD session echo rx interval
  interval          Configure BFD session interval parameters
  ipv4              Ipv4 sessions
  ipv6              Ipv6 sessions
  slow-timer        Configure slow mode timer for sessions
  startup-timer     Configure Delayed Start Up timer for sessions
```
interface:
```
n9k-109(config)# interface ethernet 1/1
n9k-109(config-if)# bfd ?
  <CR>
  authentication    Configure BFD authentication parameters
  echo              Configure Echo function for all address families
  echo-rx-interval  Configure BFD session echo rx interval
  interval          Configure BFD session interval parameters
  ipv4              Ipv4 sessions
  ipv6              Ipv6 sessions
  neighbor          BFD neighbor configuration commands (simulate client)
  optimize          Optimize
```